### DISCUSSION TITLE
Language editing in various places

### DISCUSSION THREAD

**Discussion by kendew** [lxqt/lxqt#2171] [Q&A]
I set up Linux systems for different people and I wonder if it is possible to customize the wording of text in LXQt in various places.  For example if I invoke LXQt Session Shutdown, the text reads "Do you want to really switch off your computer?  All unsaved work will be lost..."
Absolutely nothing wrong with that, but let's say I want to word it differently?  Perhaps using the local dialect, or adding the user's name to add a personal touch, like "Kendew, do you really want to..."?  In Openbox there is no problem doing this, but is there a way in LXQt?
Sometimes, too, I set up computers for non-English speakers, so the ability to edit such text would be useful for me.

**Comment by tsujan** [comment]
Your question isn't limited to LXQt.

GUIs are written in English. You can't change their English texts unless you change their codes.

Apps can be "internationalized". All LXQt apps are so. It's up to translators to add "localization". "Internationalization" means that the app's GUI is ready for being translated (= localized). The translated texts will replace the English texts automatically if you choose a locale other than English. See LXQt Locale Configuration.

LXQt apps have been translated into many languages and many translators have contributed to their localization through Weblate. If you want to translate LXQt into a new language or complete an already existing translation, you could ask @stefonarch for guidance.

**Reply by kendew** [reply]
Thank you for your reply.  Previously I set up some computers in Japanese, which is a language I am familiar with.  Recently I've tried to set up Swedish for a friend, which is a language I don't know.  I saw that LXQt support for Swedish was over 90% complete, impressive for a language spoken by fewer people than live in some large cities.  But I am a perfectionist about such things.  I don't like to set up a computer interface that doesn't at least equal Windows, and Swedish Windows doesn't have English sprinkled here and there.  Of course, I realize the resources at Microsoft's disposal are huge, and for LXQt, which only recently reached version 1.0 to have such fantastic internationalization as it does, is outstanding and very impressive.  Still, there are some points I'd like to improve upon.
For example, take the LXQt menu, also in graphical interface.  All of the categories are rendered in Swedish.  Only "About LXQt" and "Leave" remained in English.  The actions in the Leave submenu such as "Shutdown" and "Restart" were originally in English.  However, I was able to change these by simply editing the desktop files to include a Swedish translation.  Easy.  So I was asking if it was similarly possible to do so with the Shutdown dialog.
In Openbox it is very easy to do so.  Since I use an app called obmenu-generator to render a dynamic menu, all I had to do was edit that app's file, schema.pl, to say anything I wanted to as a shutdown message, similarly GUI.  Since Swedish is a language that I don't know, I set up a Swedish Windows 10 in a VM and observed the language there.  
So if there are files I could edit, I only need to know the name of those files.  If anyone can teach me the names, please do so.  Before I was aware of LXQt I would set up Openbox to function as a minimalistic DE.  When I became aware of LXQt, I was amazed that the interface looked pretty much exactly like what I was setting up, and I thought, these guys are thinking like me.  But on this point maybe we differ.  I think it's a plus to be able to customize not only things like icons or window transparencies but language as well.  I'm thinking not only of language translation but language personalization, just like we personalize colors and themes.

**Reply by tsujan** [reply]
> All of the categories are rendered in Swedish. Only...

The LXQt team really appreciates the works of translators but can't correct or complete their works. Everything depends on translators in the area of localization. Like developers, they dedicate their free times to LXQt for free. If the work of a translator is incomplete or has problems, another translator might want to complete or correct it, or the same translator might find time and improve it.

> In Openbox it is very easy to do so...

What you may see in Openbox is a specific property of a specific app. String customization is out of question for a DE like LXQt — and also for most apps that aren't related to LXQt.

BTW, Openbox isn't an LXQt component. It can just be used as the window manager with LXQt.

**Comment by stefonarch** [comment]
It can be achieved using qt5-translator, only for english you should choose some special english and 0% translated  like en_As, then you  load the translation from source code and change items to whatever you like and release them (*.qm file). 
As every update of LXQt those file (also .desktop files) will be overwritten so  you should save them not in the system but locally in ~/.local/share/lxqt/translations/application_name (same as  in /usr/share/lxqt/translations/), they will be used first if exists.
